airflow/decorators/base.py

```diff
@@ -230,7 +230,7 @@ def _handle_output(self, return_value: Any, context: Context, xcom_push: Callabl
         if isinstance(return_value, Dataset):
             self.outlets.append(return_value)
         if isinstance(return_value, list):
-            [self.outlets.append(item) for item in return_value if isinstance(item, Dataset)]
+            [self.outlets.append(item) for item in return_value if isinstance(item, Dataset)]  # type: ignore[func-returns-value]
```

Comment:
Let’s use a normal for-loop for this. Using a list comprehension in this context is not common in the code base and may cause misunderstandings.

Reply:
I think the list comprehension is more clear than a normal for loop and the for loop is needlessly complicated.

Reply:
Okay I agree that it is not consistent with the rest of the code in this file.